operators/vision/draw_bounding_box.cc

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Note: make sure the input image has the format BGR
+
+#include "draw_bounding_box.hpp"
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <gsl/narrow>
+#include <gsl/span>
+#include <unordered_map>
+#include <vector>
+
+namespace ort_extensions {
+namespace {
+constexpr std::array<std::array<uint8_t, 3>, 10> KBGRColorMap = {{
+    {{0, 0, 255}},      // Red
+    {{0, 255, 0}},      // Green
+    {{255, 0, 0}},      // Blue
+    {{255, 255, 0}},    // Cyan
+    {{255, 0, 255}},    // Magenta
+    {{0, 0, 128}},      // Dark Red/Maroon
+    {{0, 128, 0}},      // Dark Green/Lime
+    {{128, 0, 0}},      // Dark Blue/Navy
+    {{0, 0, 0}},        // Black
+    {{128, 128, 128}},  // Gray
+}};
+
+// Used to store image data, width, height, and number of channels
+struct ImageView {
+  gsl::span<uint8_t> data;
+  int64_t height;
+  int64_t width;
+  int64_t channels;
+};
+
+// To represent Boxes, [num of boxes, box_info]
+class BoxArray {
+ private:
+  void SortBoxesByScore(const float* data) {
+    boxes_by_score_.resize(NumBoxes());
+    for (size_t i = 0; i < boxes_by_score_.size(); ++i) {
+      boxes_by_score_[i] = gsl::make_span(data + i * ShapeAtDim(1), ShapeAtDim(1));
+    }
+
+    std::sort(boxes_by_score_.begin(), boxes_by_score_.end(),
+              [](const gsl::span<const float>& first, const gsl::span<const float>& second) {
+                return first[4] > second[4];
```

Comment:
nit: ShapeAtDim() doesn't seem to provide much value given it's a private method and the code could just be shape_[dim] without losing any readability.

nit: avoid magic numbers. define a constant for the index of the score in the data so '4' has some context.

Reply:
Sure.